src/main/java/com/autotune/analyzer/recommendations/engine/RecommendationEngine.java

```diff
@@ -316,20 +314,27 @@ public KruizeObject prepareRecommendations(int calCount) {
     public void generateRecommendations(KruizeObject kruizeObject) {
 
         for (K8sObject k8sObject : kruizeObject.getKubernetes_objects()) {
-            for (String containerName : k8sObject.getContainerDataMap().keySet()) {
-                ContainerData containerData = k8sObject.getContainerDataMap().get(containerName);
+            if (k8sObject.getContainerDataMap().isEmpty()) {
```

Comment:
Need to check for `namespace` type here and only then call  `generateNamespaceRecommendations` else if for some reason `containerDataMap` is empty we end up calling namespace which might not be correct

Reply:
Updated.

Reply:
@shekhar316 - Please update documentation as well

Reply:
@khansaad - Can you review the type changes done by Shekhar.